theano/tensor/signal/downsample.py

```diff
@@ -694,10 +694,10 @@ def c_code_cache_version(self):
         return (0, 7)
 
 class DownsampleFactorMaxGradGrad(Op):
-    __props__ = ('ds', 'ignore_border', 'st')
+    __props__ = ('ds', 'ignore_border', 'st', 'padding')
 
     @staticmethod
-    def out_shape(imgshape, ds, ignore_border=False, st=None):
+    def out_shape(imgshape, ds, ignore_border=False, st=None, padding=(0, 0)):
```

Comment:
This method seem a copy of DownsampleFactorMax.out_shape. Why not call it directly? Both of them are static method, so this is easy and remove duplicate code.